src/WallpaperEngine/Core/Core.cpp

```diff
@@ -58,4 +58,14 @@ irr::video::SColor Core::atoSColor (const char *str)
 irr::video::SColor Core::atoSColor (const std::string& str)
 {
     return Core::atoSColor (str.c_str ());
-}
\ No newline at end of file
+}
+
+nlohmann::detail::iter_impl<nlohmann::json> jsonFindValueRequired (nlohmann::json *data, const char *key, const char *notFoundMsg)
```

Comment:
I would use a reference instead of a pointer for the data parameter, as we're using iterators the value should never be null. That would also allow for a bit cleaner syntax in places like CScene, where &(*general_it) would be *general_it instead.

The return type should be the nlohmann::json::iterator instead of directly using the iter_impl, this makes the usage similar to the standard library and easier to understand for people that are new to the project.

It would also be good to cleanup the #include <nlohmann/json.hpp> scattered through the codebase wherever possible.

Once you've had time to change those three things I can merge the PR into the branch.